ports/stm32/i2c.c

```diff
@@ -30,11 +30,11 @@
 
 #if MICROPY_HW_ENABLE_HW_I2C
 
-#define I2C_POLL_TIMEOUT_MS (50)
+#define I2C_POLL_TIMEOUT_MS(t) ((t)?t:50)  // Use provided timeout if set, or 50ms by default
```

Comment:
It'd be more efficient to do this "default check" when timeout is provided, rather than each time it's used.  So I'd suggest to remove this macro completely and just use the timeout variable as-is in the functions below.